android-components/components/service/contile/src/main/java/mozilla/components/service/contile/ContileTopSitesProvider.kt

```diff
@@ -71,8 +69,8 @@ class ContileTopSitesProvider(
      */
     @Throws(IOException::class)
     override suspend fun getTopSites(allowCache: Boolean): List<TopSite.Provided> {
-        val cachedTopSites = if (allowCache && !isCacheExpired()) {
-            readFromDiskCache()?.topSites
+        val cachedTopSites = if (allowCache) {
+            getCachedTopSitesIfValid()
```

Comment:
It might be worth updating the comment for this function to reflect our new understanding of the fallback protocol and mention that if the server is experiencing an outage the maximum cache age will be determined by it.